### ISSUE TITLE
Kill child process silently 

### ISSUE THREAD

**Issue by ninox92** [fluent-ffmpeg/node-fluent-ffmpeg#541]
I was wondering if there is a way to silently kill a ffmpeg process?
I'm using a RTSP live stream to feed a RTMP server on a user request.
If there are no open connections left I want to close the stream using the kill() function.

**Comment by njoyard** [comment]
Yes, the ffmpeg command has a .kill() method.

**Comment by ninox92** [comment]
On .kill(), line 171 (processor.js) `ffmpegProc.on('exit', function(code, signal)` is triggerd which throws an Error on exit. `handleExit();` without parameters is never reached because proto.kill signal is 'SIGKILL' by default.

Correct me if i'm wrong.

**Comment by njoyard** [comment]
You're right, but this is the way to proceed. Killing ffmpeg makes it exit with an error status, so that's what fluent-ffmpeg reports. The 'end' event is only used when ffmpeg exits normally.

**Comment by dyeoman2** [comment]
The approach below allowed me kill an RTSP to RTMP live stream when a kill criteria was met. Hope it helps!


```
const ff = ffmpeg()

ff.input(rtspUri)
  .outputOptions(['-c copy', '-acodec aac', '-f flv'])
  .save('rtmp://127.0.0.1:1935/live/test')
  .on('progress', (p) => {
    if (killCriteria) {
      ff.kill()
      console.log('ffmpeg > killed')
    }
    console.log('ffmpeg > in progress')
  })
  .on('end', () => {
    console.log('ffmpeg > completed')
  })
  .on('error', (err) => {
    console.log(`ffmpeg > error: ${err.message}`)
  })
```
